src/harness/unittests/matchFiles.ts

```diff
@@ -95,6 +95,44 @@ namespace ts {
         assert.deepEqual(actual.errors, expected.errors);
     }
 
+    function validateMatches(expected: ts.ParsedCommandLine, json: any, host: ParseConfigHost, basePath: string, existingOptions?: CompilerOptions, configFileName?: string, resolutionStack?: Path[]) {
+        {
+            const jsonText = JSON.stringify(json);
+            const result = parseJsonText(caseInsensitiveTsconfigPath, jsonText);
+            const actual = ts.parseJsonSourceFileConfigFileContent(result, host, basePath, existingOptions, configFileName, resolutionStack);
+            for (const error of expected.errors) {
+                if (error.file) {
+                    error.file = result;
+                }
+            }
+            assertParsed(actual, expected);
+        }
+        {
+            const actual = ts.parseJsonConfigFileContent(json, host, basePath, existingOptions, configFileName, resolutionStack);
+            expected.errors = map(expected.errors, error => {
+                return <Diagnostic>{

```

Comment:
This is a gotcha in the type system, see https://palantir.github.io/tslint/rules/no-object-literal-type-assertion/. The cast doesn't appear necessary anyway.